packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -1,8 +1,89 @@
 import React from 'react';
-import { Image } from 'react-native';
 
-export default function ExpoImage({ source, ...props }) {
-  const resolvedSource = source ?? {};
+import { ImageContentPosition, ImageProps, ImageSource } from './Image.types';
+import { resolveContentFit, resolveContentPosition } from './utils';
 
-  return <Image {...props} source={resolvedSource} />;
+const resolveAssetSource = (source: ImageSource | string | number | undefined | null) => {
+  if (source == null) return null;
+
+  if (typeof source === 'string') {
+    return { uri: source };
+  }
+  if (typeof source === 'number') {
+    return { uri: String(source) };
+  }
+
+  return source;
+};
+
+const ensureUnit = (value: string | number) => {
+  const trimmedValue = String(value).trim();
+  if (trimmedValue.endsWith('%')) {
+    return trimmedValue;
+  }
+  return `${trimmedValue}px`;
+};
+
+const getObjectPositionFromContentPosition = (contentPosition?: ImageContentPosition) => {
+  const resolvedPosition =
+    typeof contentPosition === 'string' ? resolveContentPosition(contentPosition) : contentPosition;
+  if (!resolvedPosition) return undefined;
+  if (!('top' in resolvedPosition || 'bottom' in resolvedPosition)) {
```

Comment:
I think this would be less error-prone, especially when someone passes `null` or `undefined` in the object – they are also not valid and should default to `50%` but `in` would be `true` in that case.
```suggestion
  if (resolvedPosition.top == null || resolvedPosition.bottom == null) {
```

Reply:
Unfortunately this breaks TS:
<img width="909" alt="image" src="https://user-images.githubusercontent.com/5597580/206706939-15f3867d-fe37-43ef-8196-449f3dafcd88.png">